packages/elements/src/react/common/form/otp.tsx

```diff
@@ -88,173 +87,70 @@ const OTPInputStandard = React.forwardRef<HTMLInputElement, Omit<OTPInputProps,
-
     const [selectionRange, setSelectionRange] = React.useState<SelectionRange>(props.autoFocus ? ZERO : OUTSIDE);
-    const isFocused = useFocus(innerRef);
-
-    // Attach this component's UI state "selectionRange" to the native .select() to update it on focus changes
-    React.useEffect(() => {
-      const element = innerRef.current;
-
-      if (!element) return;
-
-      const originalSelect = element.select.bind(element);
-      element.select = () => {
-        originalSelect();
-        setSelectionRange([0, element.value.length]);
-      };
-    }, []);
-
-    // Continuously update the selection range when the input is focused
-    React.useEffect(() => {
-      if (!isFocused) return;
 
-      const interval = setInterval(() => {
-        if (innerRef.current && document.activeElement === innerRef.current) {
-          setSelectionRange([innerRef.current.selectionStart, innerRef.current.selectionEnd]);
-        }
-      }, 50);
+    // This ensures we can access innerRef internally while still exposing it via the ref prop
+    React.useImperativeHandle(ref, () => innerRef.current as HTMLInputElement, []);
 
-      return () => {
-        clearInterval(interval);
-      };
-    }, [isFocused]);
+    // A layout effect is used here to avoid any perceived visual lag when changing the selection
+    React.useLayoutEffect(() => {
+      if (document.activeElement !== innerRef.current) {
+        return;
+      }
+      setSelectionRange(cur => selectionRangeUpdater(cur, innerRef));
+    }, [props.value]);
 
     // Fire the requestSubmit callback when the input has the required length and autoSubmit is enabled
     React.useEffect(() => {
-      if (previousValue === undefined) return;
-
-      if (value !== previousValue && previousValue.length < length && value.length === length && autoSubmit) {
+      if (String(props.value).length === length && autoSubmit) {
         innerRef.current?.form?.requestSubmit();
       }
-    }, [previousValue, value, length, autoSubmit]);
-
-    const _select = React.useCallback(() => {
-      if (!innerRef.current) return;
-
-      const [start, end] = transformSelectionRange({ innerRef, length, value });
-      setSelectionRange([start, end]);
-    }, [innerRef, value, length]);
-
-    const _keyDown = React.useCallback(
-      (e: React.KeyboardEvent<HTMLInputElement>) => {
-        transformKeydownEvent({ e, innerRef, value });
-      },
-      [innerRef, value],
-    );
+    }, [props.value, length, autoSubmit]);
 
     return (
       <div
-        data-otp-input-segmented-wrapper
+        data-otp-input-wrapper
         style={wrapperStyle}
       >
         {/* We can't target pseudo-elements with the style prop, so we inject a tag here */}
-        <style>
-          {`
-        input[data-otp-input-segmented]::selection {
+        <style>{`
+      input[data-otp-input]::selection {
         color: transparent;
-          background-color: transparent;
```

Comment:
There was probably a good reason to use `transparent` 👀. Can add back!

Reply:
`none` is invalid:
![CleanShot 2024-03-22 at 09 21 53](https://github.com/clerk/javascript/assets/16143594/2e852150-13b0-4eb5-8f6e-cf6997c7f6a0)

Reply:
perfect! 😄